docs/design_docs/011-historacle-pricing.md

```diff
@@ -53,18 +54,18 @@ We define two Maximum values, which correspond to the most we will store of a me
 
 ### Historic Data Table
 
-| Name           | Description                                                          | How often it is recorded      | When it is pruned                     |
-| -------------- | -------------------------------------------------------------------- | ----------------------------- | ------------------------------------  |
-| `Price Stamp`  | A price for an asset recorded at a given block                       | Every `Historic Stamp Period` | After `Maximum Price Stamps` is met.  |
-| `Median Stamp` | Of a given asset & block, the median of the stored `Price Stamps`    | Every `Median Stamp Period`   | After `Maximum Median Stamps` is met. |
+| Name           | Description                                                       | How often it is recorded      | When it is pruned                     |
+| -------------- | ----------------------------------------------------------------- | ----------------------------- | ------------------------------------- |
+| `Price Stamp`  | A price for an asset recorded at a given block                    | Every `Historic Stamp Period` | After `Maximum Price Stamps` is met.  |
+| `Median Stamp` | Of a given asset & block, the median of the stored `Price Stamps` | Every `Median Stamp Period`   | After `Maximum Median Stamps` is met. |
 
 ### Proposed API
 
 Modules will have access to the following `keeper` functions from the `x/oracle` module:
 
-- `HistoricMedian(denom) (sdk.Dec, error)` returns the median price of an asset in the last `Pruning Period`
-- `WithinHistoricDeviation(denom) (bool, error)` returns whether or not the current price of an asset is within the `Standard Deviation around the Median`.
-- `MedianOfMedians(denom string, blockNum int) sdk.Dec` returns the Median of the all the Medians recorded within the past `blockNum`.
+- `HistoricMedian(denom) ([]sdk.Dec, error)` returns list median prices of an asset in the last `Pruning Period`
```

Comment:
do we need this to return `[]sdk.Dec`?

Reply:
Should it be `HistoricMedians` then instead of `HistoricMedian`?

Reply:
If the response includes multiple medians should it also include a block number?

Reply:
@toteki need your input here

Reply:
We need to return something. I think we should return medians, and let the client decide.

Reply:
from leverage side, we'll probably be using the advanced functions (`Min/Max/Median` of medians) rather than the raw slice (or single most recent) returns. With that in mind, this one can do either and I'm fine.

Reply:
> We need to return something. I think we should return medians, and let the client decide.

I agree with this comment and we think we should re-visit this topic when moving the historacle feature over to Ojo. I don't see much difference in computing things like min/max/median_of_medians on the oracle side vs on the leverage side except we have to update proto files when the calculation changes on the oracle side. Having a single endpoint that returns X number of medians in the past could be enough.

That said, let's leave it like this for now in favor of getting v1 of this feature out. `this one can do either and I'm fine`

Reply:
>  I don't see much difference in computing things like min/max/median_of_medians on the oracle side vs on the leverage side except we have to update proto files when the calculation changes on the oracle side. Having a single endpoint that returns X number of medians in the past could be enough.

The difference is that returning the values gives more flexibility, while complexity is the same (we don't save any CPU cycles).